microprofile/WFLY-14932_mp_reactive_messaging_kafka_api.adoc

```diff
@@ -0,0 +1,105 @@
+** `topic` - the Kafka topic the `Message` was received on.
+** `partition` - the Kafka partition the `Message` was received on.
+** `timestamp` - the timestamp of the `Message`.
+** `timeStampType` - the `org.apache.kafka.common.record.TimestampType` of the message.
+** `offset` - the Kafka offset of the Kafka record represented by the `Message`.
+** `headers` - An instance of `org.apache.kafka.common.header.Headers`. They have no meaning to the Kafka broker, which just forwards them on to the consumers. These can be thought of attachments/metadata on the `Message` to be used as additional information in the producer to consumer, and are dealt with on the application level. For example, a header could indicate which data centre was the origin of a message, and consumers are free to deal with this information or ignore it.
+* https://github.com/smallrye/smallrye-reactive-messaging/pull/1272/files#diff-097e0fa8873ba4f19a46c3229a185b93c2f12507785c8bcbb7b0a360ec66ee65[OutgoingKafkaRecordMetadata] contains a builder to set data to influence how Kafka sends a `Message`> This includes:
+** `key` - the key of the Kafka record represented by the `Message`.
+** `topic` - generally users should configure the Kafka topic for a stream via the application MicroProfile Config. However, in some cases they may want to specify the topic dynamically, for example depending on the contents of the data in the message that is being sent.
+** `partition` - the Kafka partition the `Message` was received on.
+** `timestamp` - the timestamp of the `Message`.
+** `timeStampType` - the `org.apache.kafka.common.record.TimestampType` of the message.
+** `offset` - the Kafka offset of the Kafka record represented by the `Message`.
+** `headers` - Allows us to set `org.apache.kafka.common.header.Headers` as explained above.
+
+
+
+=== Nice-to-Have Requirements
+* If the SmallRye Reactive Messaging Kafka User API evolves in such a way that it brings in things we don't want in WildFly, it would be good to come up with a mechanism to give an error if a user tries to use them. We will not be able to do this as compile time checks, so we would need to do this at runtime and document.
+* I would like to look into porting the following to the SmallRye Reactive Messaging Kafka user API (time permitting, and in order of priority):
+** https://github.com/smallrye/smallrye-reactive-messaging/blob/3.5.0/smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/DeserializationFailureHandler.java[DeserializationFailureHandler] - Seems good to handle issues with deserialization for records. Was requested by a customer for Quarkus.
+** KafkaRecord - A simple wrapper around message payload and the Kafka key, and easier to use than using the IncomingKafkaRecordMetadata and OutgoingKafkaRecordMetadata in the hard requirements. I think this will make it into the Kafka user API regardless of if we want it or not.
```

Comment:
Neither of these to got into the current MR, correct?

Reply:
Correct

Reply:
Thinking whether we shouldn't rather move these unfinished tasks to some section like `Possible future enhancements` so it is a bit clearer that these didn't get in with this particular RFE?